### ISSUE TITLE
[FEATURE] add option to move window buttons to the left side

### ISSUE THREAD

**Issue by joske** [pop-os/cosmic-epoch#640]
People who also use macOS are used to having them on the left side, and by muscle memory we reach to that side automatically. It slows down the user if they need to think to go to the other side.

All linux desktop environments allow this, heck, it's even possible in GNOME :wink:

**Comment by p134c0d3** [comment]
I second this. I'm used to the buttons being on the left side (how I have them in Plasma, prior to that macOS), and it's really throwing me off having them only available on the right side.

**Comment by Cavva79** [comment]
I use Gnome Tweaks to move those buttons:
![image](https://github.com/user-attachments/assets/e8ff68de-6108-4465-97ad-26da008b2552)
Window Titlebars -> Placement -> Left
![image](https://github.com/user-attachments/assets/6134c38b-5481-4a3d-b200-1111b9fe39fe)

**Comment by joske** [comment]
oh wow that works in cosmic 😮

**Comment by joske** [comment]
hmm, that only works for GNOME/gtk apps, not cosmic native ones

**Comment by awilkins** [comment]
It's good for ergonomic reasons too - you close an app, what's the most likely next thing you're going to do? Open another one! Putting the close button top left near to the "applications" button means fewer mouse miles. I remember a blogpost by the Unity team at Canonical where they had installed pointer tracking software and actually studied this.

**Comment by NatFletch** [comment]
Agreed. A good chunk of Linux users like the decorations on the left and this would improve cosmic's customization options even more.

**Comment by siggjen** [comment]
I always prefer close button on the left since my Amiga upbringing. If I can't do it here I guess I'll fade back to Plasma

**Comment by Mann-Servante** [comment]
So, Cosmic has officially been released.
Any updates on this basic feature?

**Comment by leviport** [comment]
It's still planned.

**Comment by loissascha** [comment]
also hoping for this feature

**Comment by davidsousarj** [comment]
This is the only thing keeping me from using Pop Os on a daily basis.

**Comment by dimparf** [comment]
And for me, this is the only thing that prevents me from using Pop Os every day.

**Comment by paulneibarger** [comment]
Hope this isn't forgotten

**Comment by sousaakira** [comment]
I’d also really love to see this implemented.

I’ve been using COSMIC and I really like the direction the project is taking, but having the window controls on the right is still one of the things that feels strange after years of using macOS and other desktop environments with the controls on the left.

It would be great if this could eventually become a simple user preference, for example:

* Window buttons: Left / Right
* Button order: Close / Minimize / Maximize (or customizable)

I also noticed that the workaround mentioned earlier only works for GTK applications and not for native COSMIC applications, so a compositor-level implementation would be especially useful.

Since this issue has been open for quite a while and was mentioned as “still planned” after the COSMIC release, I just wanted to add another +1 and let the developers know that this is still an important customization option for users.

If there is a technical direction already planned for implementing this, I’d also be interested in taking a look at the code and potentially helping with a PR.

Thanks for all the work on COSMIC! It’s becoming a really great desktop environment.

**Comment by JackNWhite** [comment]
I have made a [pull request](https://github.com/pop-os/libcosmic/pull/1390) to fulfil this request

**Comment by sousaakira** [comment]
> I have made a [pull request](https://github.com/pop-os/libcosmic/pull/1390) to fulfil this request

I still need that same feature for me to switch over completely to the new Cosmic.

**Comment by chemica** [comment]
I'd like this feature as well. Any news on it?

**Comment by dimparf** [comment]
@JackNWhite Is there any progress on this feature, which is very important to everyone?

**Comment by JackNWhite** [comment]
> [@JackNWhite](https://github.com/JackNWhite) Is there any progress on this feature, which is very important to everyone?

I have to apologise to the folks who are waiting for this. My initial solution received a cold reception because I made the option part of the theme, which was considered incorrect. Instead, I need to make it a Cosmic configuration. I did start looking into it, but paid work recently has overtaken everything else. Additionally, the reviewer said that the change would also need to be accepted by the Cosmic user experience people and that he was a little sceptical they would say yes at the moment.

Nonetheless, this is still on my agenda. I may get an evening to sort it out this week.
